src/php/bin/run_tests.sh

```diff
@@ -22,9 +22,19 @@ cd src/php/bin
 source ./determine_extension_dir.sh
 # in some jenkins macos machine, somehow the PHP build script can't find libgrpc.dylib
 export DYLD_LIBRARY_PATH=$root/libs/$CONFIG
+
 php $extension_dir -d max_execution_time=300 $(which phpunit) -v --debug \
   --exclude-group persistent_list_bound_tests ../tests/unit_tests
 
 php $extension_dir -d max_execution_time=300 $(which phpunit) -v --debug \
   ../tests/unit_tests/PersistentChannelTests
 
+export ZEND_DONT_UNLOAD_MODULES=1
+export USE_ZEND_ALLOC=0
+# Detect whether valgrind is executable
+if ! [ -x "$(command -v valgrind)" ]; then
+  echo 'Error: valgrind is not installed and is not executable' >&2
+  exit 1
```

Comment:
Let's do `exit 0`. Or, we can just move the main `valgrind` command below inside the `if` here, and flip the conditional. Like

```
if <valgrind is installed>
  run valgrind command
fi
```